### PR TITLE
Emit warning upon encountering multiple packages with the same name

### PR DESCRIPTION
**Pull request by danilhendrasr**
Fixes: #10669

### CODE REVIEW

**Comment by danilhendrasr** [comment]
Seems like the tests failed, I'll get back to it.

File: tests/testsuite/path.rs [hunk 1]
```diff
@@ -868,14 +868,16 @@ fn override_and_depend() {
...
+warning: skipping duplicate package `a2` found at `{}/b/../a/a2`
 [COMPILING] a2 v0.5.0 ([..])
 [COMPILING] a1 v0.5.0 ([..])
 [COMPILING] b v0.5.0 ([..])
 [FINISHED] dev [unoptimized + debuginfo] target(s) in [..]
 ",
-        )
+            p.root().to_string_lossy()
+        ))
         .run();
```
Comment (Muscraft) [inline tests/testsuite/path.rs:881]: What I put below should fix the current test failures:

```rust
    p.cargo("build")
        .cwd("b")
        .with_stderr(
            "\
warning: skipping duplicate package `a2` found at `[..]`
[COMPILING] a2 v0.5.0 ([..])
[COMPILING] a1 v0.5.0 ([..])
[COMPILING] b v0.5.0 ([..])
[FINISHED] dev [unoptimized + debuginfo] target(s) in [..]
",
        )
        .run();
```

Comment (danilhendrasr) [reply tests/testsuite/path.rs:881]: Thank you!

File: tests/testsuite/git.rs [hunk 2]
```diff
@@ -1023,6 +1023,52 @@ Caused by:
...
+        )
+        .file("src/main.rs", "fn main() {  }")
+        .build();
+
+    p.cargo("build").run();
+    p.cargo("run")
+        .with_stderr(
+            "\
+warning: skipping duplicate package `bar` found at `[..]`
+warning: skipping duplicate package `bar` found at `[..]`
```
Comment (weihanglo) [inline tests/testsuite/git.rs:1064]: Could you make the assertion texts more consistent?

```suggestion
[WARNING] skipping duplicate package `bar` found at `[..]`
[WARNING] skipping duplicate package `bar` found at `[..]`
```

File: tests/testsuite/path.rs [hunk 3]
```diff
@@ -870,6 +870,7 @@ fn override_and_depend() {
         .cwd("b")
         .with_stderr(
             "\
+warning: skipping duplicate package `a2` found at `[..]`
```
Comment (weihanglo) [inline tests/testsuite/path.rs:873]: ditto

**Review by weihanglo — COMMENTED** [review]
Thanks! 

Could you squash git commits?

**Comment by danilhendrasr** [comment]
By squashing git commits, do you mean by doing [this](https://stackoverflow.com/a/50880042)?

File: tests/testsuite/git.rs (hunk 2 above)
Comment (danilhendrasr) [reply tests/testsuite/git.rs:1064]: Yes. Didn't notice there's a pattern for matching warnings, my bad.

**Comment by weihanglo** [comment]
> By squashing git commits, do you mean by doing [this](https://stackoverflow.com/a/50880042)?

Sort of. It's a good habit to make commits meaningful for future backtraces :)
You can run `git rebase` to complete that in command line ([ref](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git))

**Comment by danilhendrasr** [comment]
Done, have I done it right though? Cause I'm not familiar with this kind of situation haha.

**Review by weihanglo — APPROVED** [review]
Looks good to me. Thanks!

I assume that this warning won't be too noisy in practice. If it turns out screaming too much or anyone in the cargo team disagrees, we can revert it at anytime.

**Pull request merged**